GPU/Common/TextureCacheCommon.cpp

```diff
@@ -2803,6 +2812,14 @@ bool TextureCacheCommon::PrepareBuildTexture(BuildTexturePlan &plan, TexCacheEnt
 	plan.w = gstate.getTextureWidth(0);
 	plan.h = gstate.getTextureHeight(0);
 
+	if (!g_DoubleTextureCoordinates) {
+		// Refuse to load invalid-ly sized textures, which can happen through display list corruption.
+		if (plan.w > 512 || plan.h > 512) {
```

Comment:
I thought Tactics Ogre or something accessed a 1024x texture at some point?  They have predictable behavior, and work fine as long as you use the top left 512x512 of them.  I don't think I've ever seen above 1024 though.

-[Unknown]

Reply:
Yeah, I had to change this to 1024 later, for this reason (was reported in FF III though).